mimic/model/identity_objects.py

```diff
@@ -62,17 +80,164 @@ def not_found(message, request):
+        ('enabled', 'enabled_key', False),
+        ('publicURL', 'public_url', u""),
+        ('internalURL', 'internal_url', u""),
+        ('adminURL', 'admin_url', u""),
+        ('RAX-AUTH:tenantAlias', 'tenant_alias', "%tenant_id%"),
+        ('versionId', 'version_id', u""),
+        ('versionInfo', 'version_info', u""),
+        ('versionList', 'version_list', u"")
+    ]
+
+    tenant_mapping = [
+        ('id', 'id_key'),
+        ('region', 'region_key'),
+        ('type', 'type_key'),
+        ('publicURL', 'public_url'),
+        ('internalURL', 'internal_url'),
+        ('adminURL', 'admin_url'),
+    ]
+
+    def __init__(self, template_dict=None):
+        """
+        Create an :obj:`EndpointTemplateStore`
+
+        :param dict template_dict (optional): optional dictionary to load the
+            Endpoint Template data from. The dictionary is simply passed to the
+            deserializer.
+        """
+        self._template_data = template_dict
+        self.id_key = None
+        self.region_key = None
+        self.type_key = None
+        self.name_key = None
+        self.enabled_key = None
+        self.public_url = None
+        self.internal_url = None
+        self.admin_url = None
+        self.tenant_alias = None
+        self.version_id = None
+        self.version_info = None
+        self.version_list = None
+
+        if self._template_data is not None:
+            self.deserialize(self._template_data)
+
+    def get_url(self, url, tenant_id):
+        """
+        Apply the tenant_id replacement to the URL.
+
+        :param text_type url: the URL to do the replacement on.
+        :param text_type tenant_id: the tenant-id to insert into the URL.
+        """
+        value_to_replace = self.tenant_alias
+        if value_to_replace is None:
+            value_to_replace = '%tenant_id%'
+
+        return url.replace(value_to_replace, tenant_id)
+
+    def serialize(self, tenant_id=None):
+        """
+        Serialize the endpoint template to a dictionary.
+
+        :param text_type tenant_id: an optional parameter to limit the
+            serialization to only values to a subset for tenant-specific
+            template information.
+        """
+        data = {}
+        if tenant_id is None:
+            for template_key, local_attribute in self.required_mapping:
+                data[template_key] = getattr(self, local_attribute)
+
+            for template_key, local_attribute, _ in self.optional_mapping:
+                value = getattr(self, local_attribute)
+                if value is not None:
+                    data[template_key] = value
+            return data
+        else:
+            data['tenantId'] = tenant_id
+            for template_key, local_attribute in self.tenant_mapping:
+                value = getattr(self, local_attribute)
+                if value is not None:
+                    data[template_key] = value
+            return data
+
+    def deserialize(self, data):
```

Comment:
It seems to me that this ought to be a classmethod that returns a new `EndpointTemplateStore` each time rather than one that has a side-effect of mutating an existing one.  Is there a use-case for multiple calls to `deserialize` on the same object?